extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigSourceConfig.java

```diff
@@ -29,8 +29,15 @@
     public Optional<List<String>> configMaps;
 
     /**
-     * Namespace to look for config maps. If this is not specified, then the namespace configured in the kubectl config context
-     * is used. If the value is specified and the namespace doesn't exist, the application will fail to start.
+     * Secrets to look for in the namespace that the Kubernetes Client has been configured for.
+     * If you use this, you probably want to enable {@code quarkus.kubernetes-config.secrets.enabled}.
+     */
+    @ConfigItem
+    public Optional<List<String>> secrets;
```

Comment:
I am wondering if we should have a warning message when this is used without `quarkus.kubernetes-config.secrets.enabled` being `true`

Reply:
I'd be careful about that. As you explained to me, we can't figure out at build time if secrets will be used or not.

Reply:
Sure yeah, you can't have a proper build time warning, but we could have a warning in the recorder which would be logged at runtime. 
I think it's better to be explicit so users will now exactly what's going on.

I can already imagine people asking on Zulip or GitHub why their configured secrets are not taking effect :)

Reply:
I see what you mean. I personally think that the Kubernetes stuff already logs a bit too much, but that's during build. I'll add this runtime warning.

Reply:
I did it, but found that in the failing case, the warning is totally lost between all the other logs :-) Will push in a sec.

Reply:
Done.